api/helpers/urls.js

```diff
@@ -1,8 +1,13 @@
+/**
+ * This function finds urls or emails that are wrapped in square brackets, i.e.
+ * [https://www.example.com|unwanted text] and [example@example|unwanted text/characters]
+ * It captures the url removing unwanted characters, and removes the emails completely
+ * the first part of the regex matches the url, the second part matches the email
+ */
+
 function cleanUrlsAndEmails(string) {
-  // this regex finds urls or emails in square brackets, and captures the url
-  // [https://www.example.com|unwanted text] and [example@example|unwanted text/characters]
   const urlAndEmailRegExPattern =
-    /\[(https?\S+?)\|.*?\]|\[[^\s@]+@[^\s@]+\.[^\s@]+\|.*?\]/g;
+    /\[(https?\S+?)\s*\|.*?\]|\[[^\s@]+@[^\s@]+\.[^\s@]+\|.*?\]/g;
```

Comment:
Is it possible to add the link that wasn't caught by the previous code into the test?  That would help to show what this change does (plus good practice as it would stop someone accidentally breaking this in the future for instance)

Reply:
Good idea Joe, I've added an additional test using the link that wasn't working.